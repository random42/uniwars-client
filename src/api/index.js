import axios from 'axios'

let server = require('./server.json');
axios.defaults.baseURL = server.base_url;

export user from './user'
export team from './team'
export uni from './uni'
export game from './game'
export chat from './chat'
export * as socket from './socket'
