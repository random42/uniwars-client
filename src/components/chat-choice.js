import React, {Component} from 'react';
import {View, ScrollView, Text, StyleSheet, FlatList, Image} from "react-native";
import {Avatar} from 'react-native-elements';
import { inject, observer } from 'mobx-react/native';
import { toJS } from 'mobx';


export class ChatChoice extends Component {

  renderItem({item}) {
    return (
      <ChatChoiceItem
        item={item}
        selected={this.props.selected}
        onPress={this.props.onPressItem}
        />
    );
  }

  _keyExtractor(item) { return item._id; }

  render() {
    let data = this.props.data;
    let {selected} = this.props;
    return (
      <FlatList
        bounces={false}
        contentContainerStyle={styles.view}
        extraData={selected}
        data={data}
        renderItem={this.renderItem.bind(this)}
        keyExtractor={this._keyExtractor}
      />
    );
  }
}


class ChatChoiceItem extends Component {

  _onPress() {
    this.props.onPress(this.props.item._id);
  }

  render() {
    let {item, selected} = this.props;
    let style = selected === item._id ? StyleSheet.flatten([styles.item,styles.selItem]) : styles.item;
    return (
      <View style={style}>
        <Avatar rounded medium source={{uri: item.image}}
          activeOpacity={0.7}
          onPress={this._onPress.bind(this)}
          />
        <Text style={styles.itemText}>{item.name}</Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  view: {
  },
  item: {
    flexDirection: 'column',
    justifyContent: 'space-around',
    alignItems: 'center',
    marginTop: 15
  },
  itemText: {
    marginTop: 5
  },
  selItem: {
    borderWidth: 1
  },
  avatarContainer: {

  },
});
