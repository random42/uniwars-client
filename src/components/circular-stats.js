import React, {Component} from 'react';
import {View, Text, StyleSheet, Dimensions} from "react-native";
import PropTypes from 'prop-types';
import Carousel from 'react-native-snap-carousel';
import { AnimatedCircularProgress } from 'react-native-circular-progress';

const PERCENTAGE_PRECISION = 2; // digits after comma
const WINDOW = Dimensions.get('window');

/*
  Statistiche circolari da scorrere orizzontalmente
*/
export class CircularStats extends Component {

  constructor(props) {
    super(props)
    this.state = {}
  }

  static propTypes = {
    stats: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string,
      hit: PropTypes.number,
      miss: PropTypes.number
    })),

  }

  static defaultProps = {
    stats: [],
  }

  componentDidMount() {

  }

  renderStat({item, index}) {
    let fill = (item.hit / (item.hit + item.miss)) * 100;
    return (
      <View key={item.name}
        style={styles.pageStyle}
      >
        <Text>{item.name}</Text>
        <AnimatedCircularProgress
          size={120}
          width={15}
          fill={fill}
          tintColor="#00e0ff"
          backgroundColor="#3d5875"
        >{(fill) => (<Text>{fill.toFixed(PERCENTAGE_PRECISION)}</Text>)}
        </AnimatedCircularProgress>
      </View>
    )
  }

  render() {
    let {containerStyle, stats} = this.props;
    return (
      <View style={containerStyle}>
      <Carousel
        data={stats}
        enableSnap={true}
        firstItem={Math.floor(stats.length/2)}
        renderItem={this.renderStat}
        itemWidth={WINDOW.width/2}
        sliderWidth={WINDOW.width}
        inactiveSlideScale={0.8}
        inactiveSlideOpacity={0.6}
      />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  pageStyle: {
    flex: 1,
    // backgroundColor: "yellow",
    borderRadius: 5,
    justifyContent: 'space-around',
    alignItems: 'center'
  },
});
