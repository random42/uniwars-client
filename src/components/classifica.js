import React, {Component} from 'react';
import {View, Text, StyleSheet, Button, FlatList, Dimensions} from "react-native";

export class Classifica extends Component {

  render() {
    return (
      <View style={[styles.view, this.props.containerStyle]}>
        <View style={{height: 30, flexDirection: 'row', justifyContent: 'space-between', backgroundColor: "purple"}}>
            <Button
              onPress={this.top5Uni}
              title="TOP 5 UNIVERSITY"
              color="#841584"
              accessibilityLabel="Learn more about this purple button"
            /> 
            <Button
              onPress={this.top5Players}
              title="TOP 5 PLAYERS"
              color="#841584"
              accessibilityLabel="Learn more about this purple button"
            /> 
        </View>
      </View>
    );
  }

  top5Uni() {
    Actions.game();
  }

  top5Players() {
    Actions.game();
  }

}



const styles = StyleSheet.create({
  view: {
    flex: 1,
    margin: 20,
    backgroundColor: 'blue',
    width: Dimensions.get('window').width * .75,
    height: Dimensions.get('window').height * .7,
    borderRadius: 20,
  }
})
