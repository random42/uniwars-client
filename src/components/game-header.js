import React, {Component} from 'react';
import {View, Text, StyleSheet, Dimensions} from "react-native";
//import {Avatar} from 'react-native-elements';
import PropTypes from 'prop-types';
import { inject, observer } from 'mobx-react/native';
import Avatar from 'react-native-user-avatar';

// let before = (      <View style={[this.props.containerStyle,styles.container]}>
//         <Avatar containerStyle={styles.avatar} large rounded source={{uri: me}}></Avatar>
//         <Text style={styles.name}>Roberto Sero</Text>
//         <Text style={styles.vs}>VS</Text>
//         <Text style={styles.name}>Son Goku</Text>
//         <Avatar containerStyle={styles.avatar} large rounded source={{uri: goku}}></Avatar>
//       </View>)

let goku = "https://images.everyeye.it/img-notizie/dragon-ball-super-toyotaro-disegna-goku-ultra-istinto-in-versione-manga-v3-329799-1280x720.jpg"
let me = "https://avatars2.githubusercontent.com/u/24995370?s=460&v=4"

const WINDOW = Dimensions.get('window');
const HEIGHT = WINDOW.height / 8
const WIDTH = WINDOW.width

export class GameHeader extends Component {

  static propTypes = {
    users: PropTypes.array
  }

  static defaultProps = {
    users: []
  }

  render() {
    return (
      <View style={[this.props.containerStyle,styles.container]}>
        <View style={styles.avatarsLevel}>
          <Avatar size={HEIGHT * 0.8} name="Roberto Sero" src={me} />
          <Text style={[styles.point,styles.point0]}>6</Text>
          <View style={styles.vs}>
            <Text style={styles.vsText}>VS</Text>
          </View>
          <Text style={[styles.point,styles.point1]}>8</Text>
          <Avatar size={HEIGHT * 0.8} name="Son Goku" src={goku} />
        </View>
        <View style={styles.namesLevel}>
          <View style={[styles.nameView,styles.leftNameView]}>
            <Text style={[styles.nameText,{marginHorizontal: 15}]}>Roberto Sero</Text>
          </View>
          <View style={[styles.nameView,styles.rightNameView]}>
            <Text style={[styles.nameText, {marginHorizontal: 15}]}>Son Goku</Text>
          </View>
        </View>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    minHeight: HEIGHT,
    height: HEIGHT
  },
  namesLevel: {
    flex: 1,
    position: 'absolute',
    width: WIDTH,
    height: HEIGHT,
    flexDirection: 'row',
    justifyContent: 'center',
    backgroundColor: 'green',
    alignItems: 'center',
  },
  avatarsLevel: {
    flex: 1,
    position: 'absolute',
    width: WIDTH - 20,
    height: HEIGHT,
    marginHorizontal: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    zIndex: 1,
    //backgroundColor: 'yellow',
    alignItems: 'center',
  },
  avatar: {
  },
  nameView: {
    backgroundColor: 'white',
    margin: 3,
    height: 50,
    width: 150,
    borderRadius: 10,
    justifyContent: 'center'
  },
  leftNameView: {
    alignItems: 'flex-end',
  },
  rightNameView: {
    alignItems: 'flex-start',
  },
  nameText: {
    fontSize: 14,
    fontWeight: '400'
  },
  vs: {
    backgroundColor: 'black',
    height: 25,
    width: 25,
    borderRadius: 25/2,
    overflow: 'hidden',
    alignItems: 'center',
    justifyContent: 'center'
  },
  vsText: {
    color: 'white',
    fontWeight: 'bold'
  },
  point: {
    //backgroundColor: 'yellow',
    //justifyContent: 'center',
    textAlign: 'center',
    height: HEIGHT * 0.4,
    width: HEIGHT * 0.4,
    top: HEIGHT * 0.35,
    fontSize: 26,
    fontWeight: 'bold',
    overflow: 'hidden',
  },
  point0: {
    left: -(HEIGHT * 0.55)
  },
  point1: {
    left: (HEIGHT * 0.55)
  }
});
