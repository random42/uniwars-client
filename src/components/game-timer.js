import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View, Text, StyleSheet, TouchableOpacity} from "react-native";
import { AnimatedCircularProgress } from 'react-native-circular-progress';

const nop = () => {
  console.log('nop')
};

export class GameTimer extends Component {

  static propTypes = {
    initialTime: PropTypes.number,
    msInterval: PropTypes.number,
    startOnMount: PropTypes.bool,
    onTimeEnd: PropTypes.func,
    onTimeStart: PropTypes.func,
    onPress: PropTypes.func,
  }

  static defaultProps = {
    initialTime: 10,
    msInterval: 1000,
    startOnMount: true,
    onTimeEnd: nop,
    onTimeStart: nop,
    onPress: nop
  }

  constructor(props) {
    super(props);
    this.interval = null;
    this.state = {
      time: Math.floor(props.initialTime)
    }
  }

  componentDidMount() {
    this.props.startOnMount && this.start()
  }

  start() {
    this.props.onTimeStart();
    this.interval = setInterval(() => {
      this.setState((prevState) => {
        let { time } = prevState;
        if (time === 0) {
          return prevState;
        }
        else {
          return { time: time - 1}
        }
      })
      if (this.state.time === 0) {
        this.stop();
        this.props.onTimeEnd();
      }
    }, this.props.msInterval);
  }

  stop() {
    clearInterval(this.interval);
  }

  setTime(time) {
    this.setState(prevState => {
      return {
        time: Math.floor(time)
      }
    })
  }

  renderTime() {
    return (
      <Text
        style={styles.timeText}
      >
        {this.state.time}
      </Text>
    )
  }

  render() {
    let {time} = this.state;
    let {initialTime, containerStyle} = this.props;
    return (
      <TouchableOpacity
        onPress={this.props.onPress}
        style={containerStyle}
      >
        <AnimatedCircularProgress
          prefill={100}
          size={100}
          width={15}
          tintColor="#00e0ff"
          backgroundColor="#3d5875"
          fill={(time/initialTime)*100}
        >
          {(fill) => {
            return this.renderTime()
          }}
        </AnimatedCircularProgress>
      </TouchableOpacity>
    );
  }
}


const styles = StyleSheet.create({
  timeText: {
    fontSize: 30,
    fontWeight: 'bold'
  }
});
