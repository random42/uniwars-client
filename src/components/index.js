export {Slide} from './slider'
export {ChatChoice} from './chat-choice'
export {RankList} from './rank-list'
export {Classifica} from './classifica'
export {MyInput} from './my-input'
export {SearchView} from './search-view'
export {Loading} from './loading'
export {CircularStats} from './circular-stats'
export {GameTimer} from './game-timer'
export {TestComponent} from './test-component'
export {GameHeader} from './game-header'
export {TextLink} from './text-link'
