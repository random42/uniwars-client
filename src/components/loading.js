import React, {Component} from 'react';
import {View, Text, StyleSheet} from "react-native";
import PropTypes from 'prop-types';
import BouncingPreloader from 'react-native-bouncing-preloader';


/**
 * Loading view, for now it's just two bouncing images
 *
 * @reactProps {Object} BouncingPreloader.propTypes see 'react-native-bouncing-preolader'
 *
 */

export class Loading extends Component {

  render() {
    return (
      <View style={styles.container}>
        <BouncingPreloader
          {...this.props}
          icons={[
          'https://www.shareicon.net/data/256x256/2016/05/04/759946_bar_512x512.png',
          'https://www.shareicon.net/data/256x256/2017/07/11/888365_cookie_512x512.png',
          ]}
          leftRotation="-680deg"
          rightRotation="360deg"
          leftDistance={-180}
          rightDistance={-250}
          speed={1200} />
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  },
});
