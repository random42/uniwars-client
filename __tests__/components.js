import renderer from 'react-test-renderer'
import React from 'react'
import * as Components from '../src/components'

describe('components', () => {
  test('circular-stats', () => {
    const { CircularStats } = Components
    const stats = [
      {
        name: 'a',
        hit: 12,
        miss: 12
      },{
        name: 'b',
        hit: 90,
        miss: 10000
      },{
        name: 'c',
        hit: 20,
        miss: 10
      }
    ]
    const tree = renderer.create(
      <CircularStats stats={stats} />
    )
    .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
