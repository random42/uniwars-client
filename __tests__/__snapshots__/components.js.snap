// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components circular-stats 1`] = `
<View
  style={undefined}
/>
`;
